controller/eks-cluster-config-handler.go

```diff
@@ -166,18 +162,29 @@ func (h *Handler) OnEksConfigRemoved(key string, config *eksv1.EKSClusterConfig)
 	}
 
 	svc := cloudformation.New(sess)
+	ec2Service := ec2.New(sess)
 
 	logrus.Infof("starting node group deletion for config [%s]", config.Spec.DisplayName)
 	waitingForNodegroupDeletion := true
 	for waitingForNodegroupDeletion {
-		waitingForNodegroupDeletion, err = deleteNodeGroups(config.Spec.DisplayName, eksService, config.Spec.NodeGroups)
+		waitingForNodegroupDeletion, err = deleteNodeGroups(config, config.Spec.NodeGroups, eksService)
 		if err != nil {
 			return config, fmt.Errorf("error deleting nodegroups for config [%s]", config.Spec.DisplayName)
 		}
 		time.Sleep(10 * time.Second)
 		logrus.Infof("waiting for config [%s] node groups to delete", config.Name)
 	}
 
+	if config.Status.ManagedLaunchTemplateID != "" {
+		logrus.Infof("deleting common launch template for config [%s]", config.Name)
+		_, err = ec2Service.DeleteLaunchTemplate(&ec2.DeleteLaunchTemplateInput{
+			LaunchTemplateId: aws.String(config.Status.ManagedLaunchTemplateID),
+		})
+		if err != nil {
```

Comment:
Is there a particular error that tends to happen often? If this is just for not found error then we can check for that specifically. We seem to retry on most other resources.

Reply:
Implemented a retry that will check for "no found" errors and retry deleting of templates that don't have "not found" errors.